src/isbn-issn.js

```diff
@@ -138,49 +191,74 @@ export default ({hyphenateISBN = false, handleInvalid = false} = {}) => {
-        if (subfield) {
-          // ISBN is valid but is missing hyphens
-          const normalizedValue = normalizeIsbnValue(subfield.value);
-          if (normalizedValue !== undefined) { // eslint-disable-line functional/no-conditional-statement
-            subfield.value = normalizedValue; // eslint-disable-line functional/immutable-data
-          } else if (handleInvalid) { // eslint-disable-line functional/no-conditional-statement
-            field.subfields.push({code: 'z', value: subfield.value}); // eslint-disable-line functional/immutable-data
-            record.removeSubfield(subfield, field);
+        field.subfields.forEach(subfield => fixField020Subfield(field, subfield));
+        return;
+      }
+      // 022 ISSN:
+      const subfield = field.subfields.find(sf => sf.code === 'a' || sf.code === 'l');
+      if (subfield && handleInvalid) { // eslint-disable-line functional/no-conditional-statement
+        // $a/$l => $y (bit overkill to add $z and remove $a/$l instead of just renaming)
+        field.subfields.push({code: 'y', value: subfield.value}); // eslint-disable-line functional/immutable-data
+        record.removeSubfield(subfield, field);
+      }
+    });
+
+    function fixField020Subfield(field, subfield) {
+      console.info(`fixField020Subfield ${subfield.code} '${subfield.value}'`); // eslint-disable-line no-console
+      if (invalidISBN(subfield.value) || subfieldRequiresHyphenation(subfield)) {
+        console.info(`  fixField020Subfield ${subfield.code} '${subfield.value}'`); // eslint-disable-line no-console
+        // ISBN is valid but is missing hyphens
+        const normalizedValue = normalizeIsbnValue(subfield.value);
+        if (normalizedValue !== undefined) { // eslint-disable-line functional/no-conditional-statement
+          // Move non-initial words from $a to $q:
+          const originalishValue = trimSpaces(subfield.value);
+          if (subfield.code === 'a' && stringHasSpace(originalishValue) && !stringHasSpace(normalizedValue)) { // eslint-disable-line functional/no-conditional-statement
+            const tail = originalishValue.substring(trimSpaces(originalishValue).indexOf(' ') + 1);
+            if (tail.length) { // eslint-disable-line functional/no-conditional-statement
+              field.subfields.push({code: 'q', value: tail}); // eslint-disable-line functional/immutable-data
+            }
           }
-        }
-      } else {
-        const subfield = field.subfields.find(sf => sf.code === 'a' || sf.code === 'l');
-        if (subfield && handleInvalid) { // eslint-disable-line functional/no-conditional-statement
-          field.subfields.push({code: 'y', value: trimSpaces(subfield.value)}); // eslint-disable-line functional/immutable-data
+          subfield.value = normalizedValue; // eslint-disable-line functional/immutable-data
+
+        } else if (subfield.code === 'a' && handleInvalid) { // eslint-disable-line functional/no-conditional-statement
```

Comment:
Try to eliminate else's and else if's

Reply:
Restructuroitu ja tältä osin suoraviivaistettu koodia